### PR TITLE
Trying to upgrade webpack

### PR DESCRIPTION
**Pull request by maxime-rainville**
Naive attempt at trying to upgrade webpack

### CODE REVIEW

File: css/modules.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-const ExtractTextPlugin = require('extract-text-webpack-plugin');
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
```
Comment (maxime-rainville) [inline css/modules.js:1]: That plugin was rename

File: css/modules.js [hunk 2]
```diff
@@ -25,52 +24,61 @@ const SUPPORTED_BROWSERS = [
  * @param {string} SRC The path to the source scss files
  * @param {string} ROOT The path to the root of the project, this is so we can scope for
  * silverstripe-admin variables.scss
- * @param {boolean} useStyle Determines whether to use the style loader or extract text plugin
  * @returns {{rules: [*,*,*,*]}}
  */
-module.exports = (ENV, { FILES_PATH, SRC, ROOT }, { useStyle } = {}) => {
```
Comment (maxime-rainville) [inline css/modules.js:31]: useStyle wasn't used anywhere that I could see.

File: js/modules.js [hunk 3]
```diff
@@ -11,17 +11,23 @@ module.exports = (ENV, { MODULES, THIRDPARTY }) => {
   return {
     rules: [
       {
-        // .js and .jsx files are caught
-        test: /\.jsx?$/,
+        test: /\.m?js/,
```
Comment (maxime-rainville) [inline js/modules.js:14]: There's some newish module JS format that ship content with `*.mjs` file. That was really annoying webpack. So I had to try to hack some config in here to handle it.

File: js/modules.js [hunk 4]
```diff
@@ -40,12 +46,13 @@ module.exports = (ENV, { MODULES, THIRDPARTY }) => {
...
-          'modernizr-loader',
-          'json-loader',
-        ],
+        test: /\.modernizrrc.js$/,
+        use: [ '@sect/modernizr-loader' ]
       },
+      {
+        test: /\.modernizrrc(\.json)?$/,
+        use: [ '@sect/modernizr-loader', 'json-loader' ]
+      }
```
Comment (maxime-rainville) [inline js/modules.js:55]: I think we should maybe aim to ditch modernizr. We don't use it all that much and it's a lot of extra work to add it to our pipline.

**Comment by GuySartorelli** [comment]
Closing in favor of https://github.com/silverstripe/webpack-config/pull/53

**Pull request closed without merging**